### ISSUE TITLE
Allow subclasses as component

### ISSUE THREAD

**Issue by vp2177** [marko-js/marko#764]
### Description

I think any JavaScript class should be allowed as a component. Also any valid `import` and `export` should be allowed.

Currently the following are not allowed (using the `marko-loader` with `webpack`):

**subclassed.component.js**
```javascript
    import Superclass from './super.component'
    module.exports = class Subclassed extends Superclass { // extends doesn't work
      ...
```

**es6-exported.component.js**
```javascript
    export default class ES6_exported { // export doesn't work
      ...
```

**default-import.marko**
```javascript
    import Module from './module' // importing default doesn't work in .marko files
    ...
```

with a **module.js**
```javascript
    export default class Module {
      ...
```

instead, this **works.marko**
```javascript
    import {default as Module} from './module' // this hack works
    ...
```

Subclassing using mixin techniques also works, but I think `extends` is preferable in 2017.

    

### Context

Less developer frustration when familiar syntaxes work.

### Open Questions

Is there a technical reason for the limitations?

**Comment by ianvonholt** [comment]
Related: https://github.com/marko-js/marko/issues/665 and https://github.com/marko-js/marko/issues/619#issuecomment-287155859

**Comment by mlrawlings** [comment]
There's a number of things here, but I believe all of them have been addressed in other issues (either before or since):

> importing default doesn't work in .marko files

Fixed in #803 

> `export default class ES6_exported { // export doesn't work`

Let's continue the discussion in #850

> `class Subclassed extends Superclass { // extends doesn't work`

As Ian said, see #665 and #619.  Ideally,  to share functionality, you should use helper methods or compose components.
